kv/kvserver: TestRequestsOnFollowerWithNonLiveLeaseholder failed

kv/kvserver.TestRequestsOnFollowerWithNonLiveLeaseholder [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18008028?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18008028?buildTab=artifacts#/) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):


```
=== RUN   TestRequestsOnFollowerWithNonLiveLeaseholder
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestRequestsOnFollowerWithNonLiveLeaseholder593634506
    test_log_scope.go:76: use -show-logs to present logs inline
    client_raft_test.go:1440: 
        	Error Trace:	pkg/kv/kvserver/client_raft_test.go:1440
        	Error:      	Should be false
        	Test:       	TestRequestsOnFollowerWithNonLiveLeaseholder
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestRequestsOnFollowerWithNonLiveLeaseholder593634506
--- FAIL: TestRequestsOnFollowerWithNonLiveLeaseholder (1.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRequestsOnFollowerWithNonLiveLeaseholder.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45116

--- a/pkg/kv/kvserver/client_raft_test.go
+++ b/pkg/kv/kvserver/client_raft_test.go
@@ -1319,7 +1319,7 @@ func TestRequestsOnFollowerWithNonLiveLeaseholder(t *testing.T) {
 
 	st := cluster.MakeTestingClusterSettings()
 	// This test is specifically designed for epoch based leases.
-	kvserver.OverrideLeaderLeaseMetamorphism(ctx, &st.SV)
+	kvserver.OverrideDefaultLeaseType(ctx, &st.SV, roachpb.LeaseEpoch)
 
 	manualClock := hlc.NewHybridManualClock()
 	clusterArgs := base.TestClusterArgs{
@@ -6890,7 +6890,7 @@ func TestRaftPreVoteUnquiesceDeadLeader(t *testing.T) {
 	// This test is specifically designed for epoch based leases, as they're the
 	// only lease type for which we have quiescence.
 	st := cluster.MakeTestingClusterSettings()
-	kvserver.OverrideLeaderLeaseMetamorphism(ctx, &st.SV)
+	kvserver.OverrideDefaultLeaseType(ctx, &st.SV, roachpb.LeaseEpoch)
 	kvserver.TransferExpirationLeasesFirstEnabled.Override(ctx, &st.SV, false)
 
 	tc := testcluster.StartTestCluster(t, 3, base.TestClusterArgs{

